server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -1426,9 +1401,20 @@ export class AgenticChatController implements ChatHandlers {
             return createAuthFollowUpResult(authFollowType)
         }
 
-        return new ResponseError<ChatResult>(LSPErrorCodes.RequestFailed, err.cause.message, {
+        // Show backend error messages to the customer.
+        if (err.code === 'QModelResponse') {
+            this.#features.logging.error(`QModelResponse Error: ${JSON.stringify(err)}`)
+            return new ResponseError<ChatResult>(LSPErrorCodes.RequestFailed, err.message, {
```

Comment:
can we add a special check here to show this message when statusCode is 500?
https://github.com/aws/aws-toolkit-vscode/pull/6984/files

Reply:
Is it safe to assume 500 is throttling? Isn't this also ISE?

Reply:
you are right, right now this is a gap in our system where we couldn't differentiate bedrock throttling and real ISE. Given the fact that we just got a lot of capacity and we fallback to 3.5, the chance of real throttling happening is low so maybe it's fine to keep this as it is for now